### PR TITLE
Exasol: Add openid support for create/alter user (7.1+)

### PR DESCRIPTION
**Pull request by sti0**
### Brief summary of the change made

This adds openid authentification support for `CREATE/ALTER USER` statements introduced in 7.1.

### Are there any other side effects of this change that we should be aware of?
Nope.

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: src/sqlfluff/dialects/dialect_exasol.py [hunk 1]
```diff
@@ -2420,6 +2420,7 @@ class CreateUserSegment(BaseSegment):
             Ref("UserPasswordAuthSegment"),
             Ref("UserKerberosAuthSegment"),
             Ref("UserLDAPAuthSegment"),
+            Ref("UserOPENIDAuthSegment"),
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_exasol.py:2423]: Nit but it’s OpenID. Should we replace all instances (except the keyword itself) with that capitalisation?

File: src/sqlfluff/dialects/dialect_exasol.py [hunk 2]
```diff
@@ -2519,6 +2521,20 @@ class UserLDAPAuthSegment(BaseSegment):
     )
 
 
+@exasol_dialect.segment()
+class UserOPENIDAuthSegment(BaseSegment):
+    """user openid authentification."""
+
+    type = "openid_auth"
+    match_grammar = StartsWith(Sequence("BY", "OPENID"))
+    parse_grammar = Sequence(
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_exasol.py:2530]: The `match_grammar` adds nothing here so suggest you merge the `match_grammar` and `parse_grammar` into one `match_grammar`.

**Review by tunetheweb — CHANGES_REQUESTED** [review]

File: src/sqlfluff/dialects/dialect_exasol.py (hunk 1 above)
Comment (sti0) [reply src/sqlfluff/dialects/dialect_exasol.py]: done

File: src/sqlfluff/dialects/dialect_exasol.py (hunk 2 above)
Comment (sti0) [reply src/sqlfluff/dialects/dialect_exasol.py]: Thanks for the hint. done

File: src/sqlfluff/dialects/dialect_exasol.py [hunk 3]
```diff
@@ -2519,6 +2521,19 @@ class UserLDAPAuthSegment(BaseSegment):
     )
 
 
+@exasol_dialect.segment()
+class UserOpenIDAuthSegment(BaseSegment):
+    """user OpenID authentification."""
```
Comment (tunetheweb) [inline src/sqlfluff/dialects/dialect_exasol.py:2526]:
```suggestion
    """User OpenID authentification."""
```

**Review by tunetheweb — APPROVED** [review]
One more nit that I’ll just merge in.

**Pull request merged**